qiskit/transpiler/passes/synthesis/high_level_synthesis.py

```diff
@@ -189,16 +201,121 @@ def run(self, dag: DAGCircuit) -> DAGCircuit:
 
 
 class DefaultSynthesisClifford(HighLevelSynthesisPlugin):
-    """The default clifford synthesis plugin."""
+    """The default clifford synthesis plugin.
+
+    For N <= 3 qubits this is the optimal CX cost decomposition by Bravyi, Maslov.
+    For N > 3 qubits this is done using the general non-optimal greedy compilation
+    routine from reference by Bravyi, Hu, Maslov, Shaydulin.
+
+    This plugin is called ``clifford.default`` in ``setup.py``.
+    """
 
     def run(self, high_level_object, **options):
         """Run synthesis for the given Clifford."""
         decomposition = synth_clifford_full(high_level_object)
         return decomposition
 
 
+class AGSynthesisClifford(HighLevelSynthesisPlugin):
+    """Clifford synthesis plugin based on the Aaronson-Gottesman method.
+
+    This plugin is called ``clifford.ag`` in ``setup.py``.
```

Comment:
Instead of saying `setup.py` here it might be a bit more clear to say something like:

```suggestion
    This plugin name is :``clifford.ag`` which can be used as the key on an :class:`~.HLSConfig` object to use this
    method with :class:`~.HighLevelSynthesis`.
```

I'm just worried from the end user perspective they won't know what a ``setup.py`` is. Not that we publish these docs anywhere.

Reply:
Thanks. I have improved all the docstrings in [92356f2](https://github.com/Qiskit/qiskit-terra/pull/9399/commits/92356f2f2a9898703a96684271ba6415d79cb0a6).